airflow/utils/db.py

```diff
@@ -1666,11 +1667,14 @@ def upgradedb(
         _reserialize_dags(session=session)
     add_default_pool_if_not_exists(session=session)
     synchronize_log_template(session=session)
-    if _revision_greater(
-        config,
-        _REVISION_HEADS_MAP["2.9.0"],
-        _get_current_revision(session=session),
+    current_version = _from_revision
+    trigger_kwargs_encryption_revision = _REVISION_HEADS_MAP["2.9.0"]
+    if (
+        _from_revision != trigger_kwargs_encryption_revision
+        and _revision_greater(config, trigger_kwargs_encryption_revision, _from_revision)
+        and _revision_greater(config, current_version, trigger_kwargs_encryption_revision)
     ):
+        # _from_revision < trigger_kwargs_encryption_version <= current_version
```

Comment:
I'm a bit confused here. If `current_vesrion = _from_revision`, then when will `_from_revision < trigger_kwargs_encryption_version <= current_version` happen?

Reply:
Same here, this comment looks quite wrong here.

Reply:
Is there confusion between `from_revision` I wonder? Not sure which is intended here.